torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -358,6 +359,23 @@ void XLAGraphExecutor::SyncTensorsGraph(std::vector<XLATensorPtr>* tensors,
   }
 }
 
+std::string XLAGraphExecutor::SyncTensorsGraphDumpHlo(
+    std::vector<XLATensorPtr>* tensors, absl::Span<const std::string> devices,
+    bool sync_ltc_data) {
+  tsl::profiler::TraceMe activity("SyncTensorsGraphDumpHlo",
+                                  tsl::profiler::TraceMeLevel::kInfo);
+  SyncTensorsConfig config;
+  config.sync_ltc_data = sync_ltc_data;
+  SyncTensorCollection coll = CollectSyncTensors(*tensors, config);
+  if (coll.indices.empty()) {
+    /* Enure previous execution is complete before exiting this
+     * function */
+    TensorCollectionBarrier(&coll);
+    return "";
```

Comment:
doesn't make sense to return something more meaningful than empty string?

Reply:
My understanding is that when the tensors to be collected is empty, that means there are async execution is happening. For this case, the current synced graph should be empty, this aligns with the current handling logic in the `mark_step` workflow. So in this case we just return empty string to users. cc @JackCaoG for confirmation on my understanding and ideas on this logic.

Reply:
hmm, shouldn't it be
```
  if (!coll.indices.empty()) {
    /* Enure previous execution is complete before exiting this
     * function */
    TensorCollectionBarrier(&coll);
  } else {
    return "";
  }
```
if indicies is empty, you don't need to dump anything so there is no point of waiting for the lock.

Reply:
I was referring the logic at https://github.com/pytorch/xla/blob/83c487111978ac221df53840f9a7e91cd94335eb/torch_xla/csrc/xla_graph_executor.cpp#L1301. Is there anything wrong with the logic there?

Reply:
hmm yea.. I think `coll.indices.empty()` means there is nothing for us to sync. I need another pair of eyes, @alanwaketan wdyt? I think this is a bug.

Reply:
actually you can delete logic here, if you are dumping HLO(and stableHLO too), you will run
https://github.com/pytorch/xla/blob/83c487111978ac221df53840f9a7e91cd94335eb/torch_xla/csrc/debug_util.cpp#L122-L130

`WaitDeviceOps` will call `barrier` I think, you don't need to do barrier here.

Reply:
@alanwaketan This is a workflow similar to `XLAGraphExecutor::SyncTensorsGraph`, but returns early without going through XLA compilation and graph execution. So I decided to put it in `XLAGraphExecutor`

Reply:
@alanwaketan I see your point, it should be cleaner to call `XLAGraphExecutro::CollectSyncTensors` outside and then pass the tensors to dump util functions for lowering and StableHLO dump.